File: AndroidDomaci1-main/app/src/main/java/vezbe/domaci1/LoginActivity.java
package vezbe.domaci1;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.Toast;

public class LoginActivity extends AppCompatActivity {


    private ImageView image;
    private EditText password;
    private EditText username;
    private Button loginBtn;
    private String userSave;

    public static final String PREF_MESSAGE_KEY = "prefMessageKey";
    public static final int PREFERENCE_WRITE_REQUEST_CODE = 222;
    private boolean messageWritten = false;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);
        init();
    }

    private void init() {
        initView();
        initListener();
    }

    private void initView() {
        image = findViewById(R.id.imageLogin);
        password = findViewById(R.id.passwordLogin);
        username = findViewById(R.id.usernameLogin);
        loginBtn = findViewById(R.id.loginBtn);
    }

    private void initListener() {
        loginBtn.setOnClickListener(v -> {
            cuvanje();
//            SharedPreferences sharedPreferences = getSharedPreferences(getPackageName(), Context.MODE_PRIVATE);
//            sharedPreferences
//                    .edit()
//                    .putString(PREF_MESSAGE_KEY, username.getText().toString())
//                    .apply();
//            messageWritten = true;
//            if (messageWritten){
//                Intent intent = new Intent(this, ArticleActivity.class);
//                intent.putExtra(ArticleActivity.USER_KEY, new User(username.getText().toString()));
//                startActivityForResult(intent, PREFERENCE_WRITE_REQUEST_CODE);
//                Toast.makeText(this, "Dobro ti meni doso " + username.getText().toString(), Toast.LENGTH_SHORT).show();
//            }
////            Intent intent = new Intent(this, ArticleActivity.class);
//           // Toast.makeText(this, "Dobro ti meni doso bato" + username.getText().toString(), Toast.LENGTH_SHORT).show();
        });

    }

    private void cuvanje() {
        SharedPreferences sharedPreferences = getSharedPreferences(getPackageName(), Context.MODE_PRIVATE);
        sharedPreferences
                .edit()
                .putString(PREF_MESSAGE_KEY, username.getText().toString())
                .apply();
        messageWritten = true;
        if (messageWritten){
            Intent intent = new Intent(this, ArticleActivity.class);
            intent.putExtra(ArticleActivity.USER_KEY, new User(username.getText().toString()));
            startActivityForResult(intent, PREFERENCE_WRITE_REQUEST_CODE);
            Toast.makeText(this, "Dobro ti meni doso " + username.getText().toString(), Toast.LENGTH_SHORT).show();
        }
    }

}

File: FoodApp-main/app/src/main/java/raf/rs/rsrafprojekat2nikolapaunovic5718rn/presentation/view/activity/LoginActivity.kt
package raf.rs.rsrafprojekat2nikolapaunovic5718rn.presentation.view.activity

import android.content.Intent
import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_login.*
import raf.rs.rsrafprojekat2nikolapaunovic5718rn.R

class LoginActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)
//        init()
    }

//    private fun init() {
//        initListeners()
//    }
//
//    private fun initListeners() {
//        login_LogIn_Btn.setOnClickListener {
//            val username = login_Username_Et.text.toString()
//            val pin_str = login_Pin_Et.text.toString()
//
//            if (username.isEmpty() || pin_str.isEmpty()) {
//
//                return@setOnClickListener
//            }
//
//            val pin = Integer.parseInt(pin_str)
//
//            if (pin_str.length != 4) {
//
//                return@setOnClickListener
//            }
//
//            if (userViewModel.login(username, pin)) {
//                val intent = Intent(this, MainActivity::class.java)
//                startActivity(intent)
//                finish()
//            } else {
//                Toast.makeText(this, R.string.login_input_error, Toast.LENGTH_SHORT).show()
//            }
//
//        }
//    }

}